src/lib.rs

```diff
@@ -887,7 +887,20 @@ pub unsafe extern "C" fn path_open(
     fdflags: Fdflags,
     opened_fd: *mut Fd,
 ) -> Errno {
-    unreachable()
+    drop(fs_rights_inheriting);
+
+    let path = slice::from_raw_parts(path_ptr, path_len);
+    let at_flags = at_flags_from_lookupflags(dirflags);
+    let o_flags = o_flags_from_oflags(oflags);
+    let flags = flags_from_descriptor_flags(fs_rights_base, fdflags);
+    let mode = wasi_filesystem::Mode::READABLE | wasi_filesystem::Mode::WRITEABLE;
+
+    State::with(|state| {
+        let file = state.get_dir(fd)?;
+        let result = wasi_filesystem::open_at(file.fd, at_flags, path, o_flags, flags, mode)?;
+        *opened_fd = result;
```

Comment:
I think this'll also want to update the `Descriptor` array within `State` with the new descriptor?

Reply:
Oops, yeah. I've now implemented that, and added `fd_close` too.